### PR TITLE
CPLP-1458-app marketplace-controller-cleanup

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1458

### CODE REVIEW

File: tests/marketplace/Apps.Service.Tests/BusinessLogic/AppReleaseBusinessLogicTest.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@
 using Org.CatenaX.Ng.Portal.Backend.Offers.Library.Service;
 using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Entities;
 using Org.CatenaX.Ng.Portal.Backend.Tests.Shared;
+using Org.CatenaX.Ng.Portal.Backend.Notification.Library;
 using Xunit;
 
 namespace Org.CatenaX.Ng.Portal.Backend.Apps.Service.BusinessLogic.Tests;
```
Comment (Phil91) [inline tests/marketplace/Apps.Service.Tests/BusinessLogic/AppReleaseBusinessLogicTest.cs:36]: Please check the unit tests, they can't be executed due to 3 errors.

**Review by Phil91 — CHANGES_REQUESTED** [review]
One thing i recognized but could add in the review, in OfferReleaseData.cs the documentation isn't aligned with the code, please check.

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 2]
```diff
@@ -41,17 +44,20 @@ public class AppReleaseBusinessLogic : IAppReleaseBusinessLogic
     private readonly IPortalRepositories _portalRepositories;
     private readonly AppsSettings _settings;
     private readonly IOfferService _offerService;
+    private readonly INotificationService _notificationService;
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs:47]: fixed

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 3]
```diff
@@ -308,4 +314,148 @@ public async Task DeleteAppRoleAsync(Guid appId, Guid roleId, string iamUserId)
...
+        if (!useCaseIds.Any())
+        {
+            throw new ArgumentException($"Use Case does not exist");
+        }
+        
+        if (useCaseIds.Any(item => !Guid.TryParse(item, out _)))
+        {
+            throw new ArgumentException($"Use Case does not exist");
+        }
+        return  this.CreateAppAsync(appRequestModel);
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs]: fixed

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 4]
```diff
@@ -308,4 +314,148 @@ public async Task DeleteAppRoleAsync(Guid appId, Guid roleId, string iamUserId)
...
+        // Add app to db
+        var appRepository = _portalRepositories.GetInstance<IOfferRepository>();
+        var appId = appRepository.CreateOffer(appRequestModel.Provider, OfferTypeId.APP, app =>
+        {
+            app.Name = appRequestModel.Title;
+            app.ThumbnailUrl = appRequestModel.LeadPictureUri;
+            app.ProviderCompanyId = appRequestModel.ProviderCompanyId;
+            app.OfferStatusId = OfferStatusId.CREATED;
+        }).Id;
+
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs]: fixed

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 5]
```diff
@@ -308,4 +314,148 @@ public async Task DeleteAppRoleAsync(Guid appId, Guid roleId, string iamUserId)
...
+            }
+            if(appDetails.salesManagerId is null)
+            {
+                nullProperties.Add($"{nameof(Offer)}.{nameof(appDetails.salesManagerId)}");
+            }
+            if(appDetails.providerCompanyId is null)
+            {
+                nullProperties.Add($"{nameof(Offer)}.{nameof(appDetails.providerCompanyId)}");
+            }
+            if(appDetails.descriptionLongIsNullOrEmpty)
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs:433]: This variable name was given by Norbert

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 6]
```diff
@@ -308,4 +314,148 @@ public async Task DeleteAppRoleAsync(Guid appId, Guid roleId, string iamUserId)
...
+            {
+                nullProperties.Add($"{nameof(Offer)}.{nameof(appDetails.descriptionLongIsNullOrEmpty)}");
+            }
+            if(appDetails.descriptionShortIsNullOrEmpty)
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs:437]: This variable name was given by Norbert

File: tests/marketplace/Apps.Service.Tests/BusinessLogic/AppReleaseBusinessLogicTest.cs (hunk 1 above)
Comment (qxz2mqe) [reply tests/marketplace/Apps.Service.Tests/BusinessLogic/AppReleaseBusinessLogicTest.cs:36]: fixed

Comment (qxz2mqe) [reply tests/marketplace/Apps.Service.Tests/BusinessLogic/AppReleaseBusinessLogicTest.cs:36]: fixed

**Review by ntruchsess — APPROVED** [review]

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**